src/btree/bt_sync.c

```diff
@@ -109,44 +109,57 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
     WT_CELL_UNPACK vpack;
+    WT_PAGE_MODIFY *mod;
 
     /* Ignore root pages as they can never be deleted. */
     if (__wt_ref_is_root(ref))
         return (false);
 
-    /* Guard against marking an in-memory page to be deleted. */
-    if (ref->state != WT_REF_DISK)
-        return (false);
-
     /* Ignore internal pages, these are taken care of during reconciliation. */
     if (!__wt_ref_is_leaf(session, ref))
         return (false);
 
-    WT_ORDERED_READ(addr, ref->addr);
-    if (!__wt_off_page(ref->home, addr)) {
-        __wt_cell_unpack(session, ref->home, (WT_CELL *)addr, &vpack);
-        return (__wt_txn_visible_all(session, vpack.newest_stop_txn, vpack.newest_stop_ts));
+    /* Check for the page obsolete, if the page is modified and reconciled. */
+    mod = (ref->page != NULL) ? (ref->page->modify) : NULL;
+    if (mod != NULL && mod->rec_result == WT_PM_REC_REPLACE)
+        return (__wt_txn_visible_all(
+          session, mod->mod_replace.newest_stop_txn, mod->mod_replace.newest_stop_ts));
+
+    if (mod != NULL && mod->rec_result == WT_PM_REC_MULTIBLOCK)
+        return (__wt_txn_visible_all(
+          session, mod->mod_multi_newest_stop_txn, mod->mod_multi_newest_stop_ts));
+
+    /* Check for the obsolete from page disk address. */
```

Comment:
The comment is unclear. Does it mean:
```Check if the page is obsolete using the page disk address.``` ?

Reply:
Yes, Use that sentence.